cmd/fake-machine-controller/machine-controller.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/apiserver/pkg/util/logs"
+
+	"github.com/kubernetes-incubator/apiserver-builder/pkg/controller"
+
+	"sigs.k8s.io/cluster-api/pkg/controller/config"
+	"sigs.k8s.io/cluster-api/pkg/controller/machine"
+	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
+
+	"github.com/openshift/cluster-operator/pkg/clusterapi/fake"
+)
+
+var (
+	defaultAvailabilityZone string
+	logLevel                string
+)
+
+const (
+	controllerLogName = "awsMachine"
+	defaultLogLevel   = "info"
+)
+
+func init() {
+	config.ControllerConfig.AddFlags(pflag.CommandLine)
+	pflag.CommandLine.StringVar(&defaultAvailabilityZone, "default-availability-zone", "us-east-1c", "Default availability zone for machines created by this controller.")
```

Comment:
Is this command-line argument accepted so that scripts that are passing it in for the real AWS machine controller do not need to be modified? Or was it an oversight to retain it?

Reply:
Yes we need to have compatible flags so we can just replace the image in the deployment config template. Otherwise we have parameterize the args as well.